general/forall.hpp

```diff
@@ -42,30 +42,30 @@ const int MAX_Q1D = 14;
 // The MFEM_FORALL wrapper
 #define MFEM_FORALL(i,N,...)                             \
    ForallWrap<1>(true,N,                                 \
-                 [=] MFEM_DEVICE (int i) {__VA_ARGS__},  \
-                 [&] MFEM_LAMBDA (int i) {__VA_ARGS__})
+                 [=] MFEM_DEVICE (int i) mutable {__VA_ARGS__},  \
+                 [&] MFEM_LAMBDA (int i) mutable {__VA_ARGS__})
 
 // MFEM_FORALL with a 2D CUDA block
 #define MFEM_FORALL_2D(i,N,X,Y,BZ,...)                   \
    ForallWrap<2>(true,N,                                 \
-                 [=] MFEM_DEVICE (int i) {__VA_ARGS__},  \
-                 [&] MFEM_LAMBDA (int i) {__VA_ARGS__},\
+                 [=] MFEM_DEVICE (int i) mutable {__VA_ARGS__},  \
+                 [&] MFEM_LAMBDA (int i) mutable {__VA_ARGS__},\
```

Comment:
As far as I understand it, `mutable` applies only when capturing by copy (see https://en.cppreference.com/w/cpp/language/lambda), so on host, where we capture by reference, it is unnecessary.